File: sdk/runanywhere-swift/Sources/RunAnywhere/Capabilities/SpeakerDiarization/Models/Output/SpeakerDiarizationSpeakerInfo.swift
//
//  SpeakerDiarizationSpeakerInfo.swift
//  RunAnywhere SDK
//
//  Information about a detected speaker
//

import Foundation

// MARK: - Deprecated Typealias

/// Deprecated typealias for backward compatibility
@available(*, deprecated, renamed: "SpeakerDiarizationSpeakerInfo")
public typealias SpeakerInfo = SpeakerDiarizationSpeakerInfo

// MARK: - Speaker Info

/// Information about a detected speaker
public struct SpeakerDiarizationSpeakerInfo: Sendable {

    /// Unique identifier for the speaker
    public let id: String

    /// Optional display name for the speaker
    public var name: String?

    /// Confidence score for the speaker identification (0.0-1.0)
    public let confidence: Float?

    /// Speaker embedding vector (for comparison)
    public let embedding: [Float]?

    // MARK: - Initialization

    public init(
        id: String,
        name: String? = nil,
        confidence: Float? = nil,
        embedding: [Float]? = nil
    ) {
        self.id = id
        self.name = name
        self.confidence = confidence
        self.embedding = embedding
    }
}

// MARK: - Equatable

extension SpeakerDiarizationSpeakerInfo: Equatable {
    public static func == (lhs: SpeakerDiarizationSpeakerInfo, rhs: SpeakerDiarizationSpeakerInfo) -> Bool {
        lhs.id == rhs.id
    }
}

// MARK: - Hashable

extension SpeakerDiarizationSpeakerInfo: Hashable {
    public func hash(into hasher: inout Hasher) {
        hasher.combine(id)
    }
}

File: sdk/runanywhere-flutter/packages/runanywhere/lib/features/speaker_diarization/models/speaker_diarization_speaker_info.dart
/// Information about a detected speaker
/// Matches iOS SpeakerDiarizationSpeakerInfo from Features/SpeakerDiarization/Models/SpeakerDiarizationSpeakerInfo.swift
class SpeakerDiarizationSpeakerInfo {
  /// Unique identifier for the speaker
  final String id;

  /// Optional display name for the speaker
  String? name;

  /// Confidence score for the speaker identification (0.0-1.0)
  final double? confidence;

  /// Speaker embedding vector (for comparison)
  final List<double>? embedding;

  SpeakerDiarizationSpeakerInfo({
    required this.id,
    this.name,
    this.confidence,
    this.embedding,
  });

  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;
    return other is SpeakerDiarizationSpeakerInfo && other.id == id;
  }

  @override
  int get hashCode => id.hashCode;

  /// Create from map
  factory SpeakerDiarizationSpeakerInfo.fromJson(Map<String, dynamic> json) {
    return SpeakerDiarizationSpeakerInfo(
      id: json['id'] as String,
      name: json['name'] as String?,
      confidence: (json['confidence'] as num?)?.toDouble(),
      embedding: (json['embedding'] as List?)?.cast<double>(),
    );
  }

  /// Convert to map
  Map<String, dynamic> toJson() => {
        'id': id,
        if (name != null) 'name': name,
        if (confidence != null) 'confidence': confidence,
        if (embedding != null) 'embedding': embedding,
      };
}

/// Deprecated typealias for backward compatibility
@Deprecated('Use SpeakerDiarizationSpeakerInfo instead')
typedef SpeakerInfo = SpeakerDiarizationSpeakerInfo;
